lib/server.js

```diff
@@ -37,31 +37,57 @@ module.exports = class AlgoServer extends EventEmitter {
     } = args
 
     this.port = port
-    this.host = new AOHost({
-      aos, apiKey, apiSecret, agent, wsURL, restURL
-    })
+    // If provided with order definitions then create the algoHost instance
+    // Mostly just for backwards compatibility
+    if (aos) {
```

Comment:
We should require the user to either pass in an `aos` definition, or a `host`, otherwise starting the wss server in the constructor is dangerous if no host is set/configured.

Reply:
Yeah good point, imo I think we should fix this by making the websocket less dangerous if there isnt an active host. Its kind of a weird side effect to throw an error if there isnt a set of order definitions/algo host passed into the constructor (especially if we are providing a `setHost` function)

Reply:
Also im not sure I like the idea of confusing the `aos` variable with 2 different types (array of order definitions, algo host instance) so im going to leave that, unless you think it would cause a breaking change?